### PR TITLE
CMD-25 filter properties that form options

### PR DESCRIPTION
**Pull request by DeszkaCodes**

### CODE REVIEW

File: CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+﻿using com.cyberinternauts.csharp.CmdStarter.Tests.Common.Interfaces;
+
+namespace com.cyberinternauts.csharp.CmdStarter.Tests.Commands.GlobalOptions
```
Comment (djon2003) [inline CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs:3]: Isn't missing a last `.Filters` namespace?

This would apply to all three.

Comment (djon2003) [inline CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs:3]: Under `Tests.Commands.Options` you used `Attributes` and here `Filters`. I prefer your new name as `Attributes` used one namespace above is structured with sub folders with the name of the attribute, but not under `Options.Attributes`. What do you think?

P.S. In fact, I prefer all the way the names used in `GlobalOptions.Filters` than the ones in `Options.Attributes`. The idea is to be consistent. What do you suggest?

File: CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs
Comment (djon2003) [inline CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs:1]: Isn't missing the base test case as `OptByLackOfAttribute`, but for global options?

**Review by djon2003 — CHANGES_REQUESTED** [review]

File: CmdStarter.Tests.Commands/GlobalOptions/OptionlessCommand.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿namespace com.cyberinternauts.csharp.CmdStarter.Tests.Commands.GlobalOptions
+{
+    public sealed class OptionlessCommand : StarterCommand
```
[the remark this answers is not in the document]
Comment (DeszkaCodes) [reply CmdStarter.Tests.Commands/GlobalOptions/OptionlessCommand.cs:3]: I added this command but I couldn't find more.

File: CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+{
+    [AllOptionsExcluded]
```
[the remark this answers is not in the document]
Comment (DeszkaCodes) [reply CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs:5]: I think it should be allowed just to make sure everything works fine.

File: CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionsFilterOnlyExclude.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
+﻿using com.cyberinternauts.csharp.CmdStarter.Tests.Common.Interfaces;
+
+namespace com.cyberinternauts.csharp.CmdStarter.Tests.Commands.GlobalOptions
+{
+    public sealed class GlobalOptionsFilterOnlyExclude : IGlobalOptionsContainer, IOptByAttribute
```
[the remark this answers is not in the document]
Comment (DeszkaCodes) [reply CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionsFilterOnlyExclude.cs:5]: It's a typo I haven't noticed.

File: CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs (hunk 1 above)
Comment (DeszkaCodes) [reply CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs:3]: Yes it is, fixed.

File: CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (djon2003) [reply CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs:5]: And why wouldn't it work fine if we disallow it?

File: CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs
Comment (DeszkaCodes) [reply CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs:1]: Added.

File: CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (DeszkaCodes) [reply CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs:5]: Well we can delete this test, if you think it would be better.

File: CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs (hunk 1 above)
Comment (DeszkaCodes) [reply CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs:3]: Should I rename the `Options.Attributes` to be like `GlobalOptions.Filters`?

File: CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (djon2003) [reply CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs:5]: For now, as I said, I would keep it but add the edge case into CMD-16. This is minor hence postponing it.

File: CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs (hunk 1 above)
Comment (djon2003) [reply CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs:3]: That's what I preconize. Do you agree?

File: CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (DeszkaCodes) [reply CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs:5]: Okay!

[the remark this answers is not in the document]
Comment (djon2003) [reply CmdStarter.Tests.Commands/GlobalOptions/Filters/GlobalOptionFilterExcludeAll.cs:5]: Just for posterity, what I meant was that `[GlobalOptionsContainer]` and `[AllOptionsExcluded]` should be mutually exclusive.

I added it to the story CMD-16.

**Review by djon2003 — APPROVED** [review]
